### ISSUE TITLE
Incorrectly removing used imports

### ISSUE THREAD

**Issue by stemis** [sweepline/eslint-plugin-unused-imports#24]
I have the following file:

```
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { UserEntity } from './user.entity';

@Injectable()
export class UsersService {
    constructor(
        @InjectRepository(UserEntity)
        private readonly _usersRepository: Repository<UserEntity>,
    ) {}

    findAll(): Promise<UserEntity[]> {
        return this._usersRepository.find();
    }
}
```

After running eslint fix, it deletes `import { Repository } from 'typeorm';` while `@typescript-eslint/no-unused-vars` does not detect it as unused import originally
eslint output:
```
  3:10  error  'Repository' is defined but never used  unused-imports/no-unused-imports-ts

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

Eslint config:
```
{
    "no-unused-vars": 0,
    "@typescript-eslint/no-unused-vars": "off",
    "unused-imports/no-unused-imports-ts": "error",
}
```

This is only a single file, but there are more files throughout my whole project where this is happening.
Only after adding `"unused-imports/no-unused-imports-ts": "error",` a lot of used imports are being deleted.
When running ESLint with only:
```
{
        "no-unused-vars": 0,
        "@typescript-eslint/no-unused-vars": 2,
}
```
No unused import are detected

I'd be happy to provide more examples/code. Let me know!

**Comment by vuki656** [comment]
Also experiencing this same case.

**Comment by sweepline** [comment]
I have recreated this personally and i cant seem to find solution.

Using
```
		"@typescript-eslint/no-unused-vars": "error",
		"unused-imports/no-unused-imports-ts": "error",
```
fixes the problem for me, can you confirm? This does give the annoyance of problems being reported twice (once by each rule) though.

**Comment by sweepline** [comment]
Okay I seem to have found a way to replicate this. If the explicitly installed version of `@typescript-eslint/eslint-plugin` doesn't match the one installed by my plugin it doesnt work. I guess either I should remove the dependency from my package or be more explicit about it.

So.. After updating to the following in my `package.json`
```
    "@typescript-eslint/eslint-plugin": "^4.14.2",
    "@typescript-eslint/parser": "^4.14.2",
```
The issue goes away even with 
```
rules: {
        "@typescript-eslint/no-unused-vars": "error",
        "unused-imports/no-unused-imports-ts": "error",
}
```

**Comment by sweepline** [comment]
I have pushed version `1.0.2` which should remove the dependencies such that it uses the installed version, can you confirm this works for you?

**Comment by Farenheith** [comment]
Hello, @sweepline!
Thanks for your efforts in this issue! I just fell on the same problem. Look that, at this print, GetExampleResponse are being used as return type of exampleQuery, but it is accusing error
![image](https://user-images.githubusercontent.com/5713887/107038492-5e9ef680-679b-11eb-906c-913dc323d054.png)

I have this configuration in eslintrc:
```ts

    '@typescript-eslint/no-unused-vars': 'off',
		'unused-imports/no-unused-imports-ts': 'error',
		'unused-imports/no-unused-vars-ts': [
			'error',
			{ 'vars': 'all', 'varsIgnorePattern': '^_', 'args': 'after-used', 'argsIgnorePattern': '^_' }
		],
```

If I run the lint with the --fix option, the import got removed.
A workaround I found for this is to slightly change my method:
![image](https://user-images.githubusercontent.com/5713887/107038714-ade52700-679b-11eb-8517-a1bec8b9faa7.png)
Making a typecast in the return value seems to satisfy the lint validation.

So, I guess the problem is that the validation is not considering the function declared return type.

**Comment by Farenheith** [comment]
Btw, sorry for not telling it before, but I'm testing with the 1.0.3 version

**Comment by vuki656** [comment]
> I have pushed version `1.0.3` which should remove the dependencies such that it uses the installed version, can you confirm this works for you?
> 
> _It seems I made a mistake please use `1.0.3`_ instead!_

Just tested it with `1.0.3` and it didn't seem to fix the issue.

Let me know if you'd like a reproducible example.

Thanks for taking the time to help us.

**Comment by sweepline** [comment]
```
import { Get, Query } from "somewhere";

import { Generic } from "../resource";
import func from "../util";

export class Foo {
	@Get()
	async getPromise(@Query("params") params: string): Promise<Generic> {
		const response = await func(params);
		return response;
	}
}
```
```
$ ./node_modules/.bin/eslint scalgo/js/ui/components/Slider/x.ts --config .eslintrc.js
$ echo $?
0
```
This example gives me no problems with the following relevant packages in my `package.json`:
```
devDependencies: {
    "@typescript-eslint/eslint-plugin": "^4.14.2",
    "@typescript-eslint/parser": "^4.14.2",
    "eslint": "^7.19.0",
    "eslint-plugin-unused-imports": "^1.0.1"
}
```
and eslint rules
```
rules: {
	"@typescript-eslint/no-unused-vars": "off",
	"unused-imports/no-unused-imports-ts": "error",
	"unused-imports/no-unused-vars-ts": [
		"warn",
		{ vars: "all", varsIgnorePattern: "^_", args: "after-used", argsIgnorePattern: "^_" },
	],
},

```

**Comment by Farenheith** [comment]
@sweepline nice! Forcing the @typescript-eslint/parser seems to solve the problem!

Thank you!

**Comment by stemis** [comment]
Works like a charm now, thanks @sweepline !
